### ISSUE TITLE
Generate QR code for configured server setting in Service app 

### ISSUE THREAD

**Issue by laddha-adi** [opendatakit/opendatakit#1391]
Services should be modified to add a button that generates and display a QR code, containing information about server setting's current configuration.
Examples of how JSON Object for QR code would look like-
For anonymous configuration

```
{
  "url": "https://yoururl.appspot.com"
}
```
For username-password configuration

```
{
 "url": "https://yoururl.appspot.com",
"username":"username",
"password":"password"
}
```
**Would you like to work on this issue?** 
Yes

**Comment by laddha-adi** [comment]
@opendatakit-bot claim

**Comment by wbrunette** [comment]
@laddha-adi thanks for filing this issue as it is the next step to make it easy for users to take advantage. I am not sure Services is the right place to generate the icon. Should it be part of a desktop app? Thoughts?

**Comment by jbeorse** [comment]
I can see utility in both options.

Use Case #1: An admin sets up a server and wants to distribute the server settings to field workers. In this case I think a desktop application makes more sense. Possibly an addition to App Designer, but that integration can happen later. I think the desktop application is easier to use and more convenient for distribution of the barcode. The admin can paste in the Server URL, generate an image file, and distribute it via printing copies, emailing, SMS, WhatsApp, or some other option.

User Case #2: I am part of a team of field workers, and my phone is fully configured. A new member joins and is setting up their phone for the first time. In this case, it would be convenient for me to tap a button in Services that displays a QR code that the new member can scan to configure their settings to match mine.

In my opinion, Use Case #1 is the core use case that we should cover, and Use Case #2 is a nice addition, but not as important.

**Comment by linl33** [comment]
I agree with @jbeorse's 2 use cases. 

For use case 1, it might be more useful to put the generator function on sync-web-ui. This way the server could generate QR codes for all users (without password). 

For use case 2, could also be useful when settings up multiple devices.

**Comment by laddha-adi** [comment]
I totally agree with @jbeorse, generating QR code is not a complex task and would be an added advantage for users.

**Comment by laddha-adi** [comment]
Hello everyone, any suggestions for the UI? Like where should I place the button to generate QR code and how should I display the QR code?

**Comment by jbeorse** [comment]
I don't have strong feelings about this. My suggestion would be to make the QR button in Services create a drop down menu. One item would read "Import Settings" and would launch the scanner as it does now. One item would read "Export Settings" and would launch the generator. The generator would render the QR code and show what was stored in it. 

For this case I would also say we should only export the server URL and not the authentication credentials. Definitely not the password. Once the password is stored there should be no way for the user to ever see it again.

**Comment by clarlars** [comment]
Sorry for the late response, but I also agree with Jeff in that Use Case 1 is the more useful implementation.

**Comment by laddha-adi** [comment]
Hi, Sorry for being inactive for a long time, actually I had my university exams. 
Keeping in mind all the above views, should I still continue to work on this feature or it is not that useful as of now and we should skip it?

**Comment by wbrunette** [comment]
It's a useful feature.

**Comment by laddha-adi** [comment]
@opendatakit-bot  claim

**Comment by laddha-adi** [comment]
Alright, then I'll be submitting a pull request for this soon. :)
